plugin/storage/cassandra/spanstore/operation_names.go

```diff
@@ -67,12 +106,27 @@ func NewOperationNamesStorage(
 	}
 }
 
+func tableExist(session cassandra.Session, tableName string) bool {
+	query := session.Query(fmt.Sprintf(TableQueryStmt, tableName))
+	err := query.Exec()
+	return err == nil
+}
+
 // Write saves Operation and Service name tuples
 func (s *OperationNamesStorage) Write(serviceName string, operationName string) error {
 	var err error
+	//TODO: take spanKind from args
+	spanKind := ""
 	query := s.session.Query(s.InsertStmt)
-	if inCache := checkWriteCache(serviceName+"|"+operationName, s.operationNames, s.writeCacheTTL); !inCache {
-		q := query.Bind(serviceName, operationName)
+	if inCache := checkWriteCache(serviceName+"|"+spanKind+"|"+operationName, s.operationNames, s.writeCacheTTL); !inCache {
+		var q cassandra.Query
+		switch s.SchemaVersion {
+		case 1:
+			q = query.Bind(serviceName, spanKind, operationName)
+		case 0:
+			q = query.Bind(serviceName, operationName)
```

Comment:
I would do this switch in the constructor and just capture a function as a struct field:
```
bindQuery func(query cas.Query, service, kind, opName string)
```
Also, it would be better to encapsulate all these differences in the tableMeta object